the_wizz/pair_maker.py

```diff
@@ -123,19 +128,15 @@ def run(self, reference_catalog, unknown_catalog, use_unkn_weights=False):
             self.hdf5_writer = Pool(1)
 
         print("Starting iteration...")
-        for ref_vect, redshift, dist, ref_id in zip(ref_vects,
-                                                    redshifts,
-                                                    dists,
-                                                    ref_ids):
+        ref_obj_iter = zip(
+            reference_catalog["ra"], reference_catalog["dec"],
+            redshifts, dists, ref_ids)
+        for ref_ra, ref_dec, redshift, dist, ref_id in ref_obj_iter:
             # Query the unknown tree.
-            unkn_idxs = np.array(self._query_tree(ref_vect, unkn_tree, dist))
-
-            # Compute angles and convert them to cosmo distances.
-            matched_unkn_vects = unkn_vects[unkn_idxs]
-            cos_thetas = np.dot(matched_unkn_vects, ref_vect)
-            matched_unkn_dists = np.arccos(cos_thetas) * dist
-            matched_unkn_ids = unkn_ids[unkn_idxs]
-
+            ang_min = self.r_min / dist  # convert to angular scales
```

Comment:
These angles are naturally in radians, hence the query_shell below is not finding using the proper distances.